crates/gateway-types/src/reply.rs

```diff
@@ -41,20 +64,61 @@ pub struct Block {
     /// Version metadata introduced in 0.9.1, older blocks will not have it.
     #[serde(default)]
     pub starknet_version: StarknetVersion,
+
+    // Introduced in v0.13.1
+    #[serde(default)]
+    pub transaction_commitment: Option<TransactionCommitment>,
+    #[serde(default)]
+    pub event_commitment: Option<EventCommitment>,
+    #[serde(default)]
+    pub l1_da_mode: Option<L1DataAvailabilityMode>,
+}
+
+impl Block {
+    pub fn eth_l1_gas_price(&self) -> Option<GasPrice> {
+        self.l1_gas_price_implementation_detail
+            .map(|p| p.price_in_wei)
+            .or(self.eth_l1_gas_price_implementation_detail)
+    }
+
+    pub fn strk_l1_gas_price(&self) -> Option<GasPrice> {
+        self.l1_gas_price_implementation_detail
+            .map(|p| p.price_in_fri)
+            .or(self.strk_l1_gas_price_implementation_detail)
+    }
```

Comment:
Nice, I like how you handled this. Much simpler than the nested enum I would likely have reached for; especially considering we're allowed to simplify after v0.13.1 rolls out everywhere.